api/envoy/extensions/load_balancing_policies/common/v3/common.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+
+package envoy.extensions.load_balancing_policies.common.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/v3/percent.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.load_balancing_policies.common.v3";
+option java_outer_classname = "CommonProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/common/v3;commonv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Common configuration for two or more load balancing policy extensions]
+// [#not-implemented-hide:]
+
+// Configuration for :ref:`zone aware routing
+// <arch_overview_load_balancing_zone_aware_routing>`.
+message ZoneAwareLbConfig {
```

Comment:
Would it make sense to nest `ZoneAwareLbConfig` and `LocalityWeightedLbConfig` inside of `LocalityLbConfig`?

Reply:
> Looks like we're repeating this oneof in 3 different places. Would it make sense to define a new message in common.proto that contains this oneof, and then just have a single field of that new message type in each of these three places?

It's used to address previous comment. As said in the comment, the `oneof` repeat three times and this change eliminate this repeating.

Reply:
Right, the `LocalityLbConfig` message is good.  What I'm suggesting here is to move the `ZoneAwareLbConfig` and `LocalityWeightedLbConfig` messages to be inside of `LocalityLbConfig`, since they're not used anywhere else.

Reply:
Hah, get it.